server/src/main/kotlin/io/provenance/invoice/services/EventHandlerService.kt

```diff
@@ -43,12 +44,10 @@ class EventHandlerService(
             // Payments should only be allowed for oracle-approved invoices
             InvoiceStatus.APPROVED,
         )
-
-        private val HANDLER_THREAD_POOL = Executors.newFixedThreadPool(10)
     }
 
     fun handleEvent(event: StreamEvent, isRetry: Boolean = false) {
-        try {
+        Either.catch {
```

Comment:
Mostly just playing around with `Either` here.  I think I like the syntax, but it doesn't really give us anything